server/channels/app/notification_push.go

```diff
@@ -56,10 +56,11 @@ type PushNotification struct {
 	explicitMention    bool
 	channelWideMention bool
 	replyToThreadType  string
+	messageOnly        bool
 }
 
 func (a *App) sendPushNotificationSync(c request.CTX, post *model.Post, user *model.User, channel *model.Channel, channelName string, senderName string,
-	explicitMention bool, channelWideMention bool, replyToThreadType string) *model.AppError {
+	explicitMention bool, channelWideMention bool, replyToThreadType string, messageOnly bool) *model.AppError {
```

Comment:
Not that I can think of, it would require rewriting all the dependent upstream and downstream paths. I wanted to keep the changes as minor as possible, but it does add on even more cruft (you can see the cruft from CRT and the id generic notifications).
If we want to go that far, we may as well add an option to bypass the generic message generation so that plugins could send generic messages like "Sender is inviting you to a call" on servers less than `full` but more permissive that `idLoaded`.

Reply:
> If we want to go that far, we may as well add an option to bypass the generic message generation so that plugins could send generic messages like "Sender is inviting you to a call" on servers less than full but more permissive than idLoaded.

From the plugin implementer side that sounds much simpler although more powerful, meaning some extra care will be needed to make sure we respect related config settings. 

That said, I don't want to block this PR so I'll let you decide whether the extra effort is worth it.

Reply:
Maybe a `pluginOverride` field on `PostNotification` and `app.PushNotification` which would send the raw `post.Message` as the push message. 
That would help us respect the `idLoaded` setting, and still keep the processing benefits of the code path.
I know allowing a plugin to send a `model.PushNotification` would be more powerful and easier, but I still think that puts too much work onto the plugin.

Reply:
(note: continued conversation on MM)